### PR TITLE
And system prune feature for v2.

### PR DESCRIPTION
**Pull request by sujil02**
Adds podman system prune for v2.
Refactoring for code reuse from pods containers images and volume prune.
Adds and enables testcases to support the added feature.

Signed-off-by: Sujil02 <sushah@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
lgtm
@baude @mheon @giuseppe @TomSweeneyRedHat @QiWang19 @jwhonce PTAL

File: cmd/podman/system/prune.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: pruneCommand,
+		Parent:  systemCmd,
+	})
+	flags := pruneCommand.Flags()
+	flags.BoolVarP(&force, "force", "f", false, "Do not prompt for confirmation.  The default is false")
+	flags.BoolVarP(&pruneOptions.All, "all", "a", false, "Remove all unused data")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/system/prune.go:46]: this might be a @mheon question, should this be "unused data" or "unused data and objects"?

**Comment by TomSweeneyRedHat** [comment]
One question, LGTM otherwise.  I'm fine with merging now and adjusting later if required.

**Pull request merged**